### PR TITLE
[+] BO: added a tag management of product categories

### PR DESCRIPTION
**Pull request by mickaelandrieu**
Description and steps described in forge ticket
## Forge Ticket (optional)

http://forge.prestashop.com/browse/BOOM-405

boom405

### CODE REVIEW

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ $(document).ready(function() {
 	displayFieldsManager.init();
 	seo.init();
 	tags.init();
+  productCategoriesTags.init();
```
Comment (mickaelandrieu) [inline admin-dev/themes/default/js/bundle/product/form.js]: remove extra space

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 2]
```diff
@@ -2049,6 +2050,78 @@ var tags = (function() {
 	};
 })();
 
+/**
```
Comment (mickaelandrieu) [inline admin-dev/themes/default/js/bundle/product/form.js]: this should be moved on his own file, and "required" in `product.js`

File: src/Adapter/Category/CategoryDataProvider.php [hunk 3]
```diff
@@ -69,4 +70,32 @@ public function getAllCategoriesName($root_category = null, $id_lang = false, $a
...
         return $categories;
     }
+
+    /**
+     * return a simple array id/fullpath/name of categories for a specified product
+     * @param Product $product
+     *
+     * @return array Categories
+     */
+    public function getCategoriesByProduct(ObjectModel $product)
```
Comment (mickaelandrieu) [inline src/Adapter/Category/CategoryDataProvider.php]: using `Product` instead make PrestaShop cry : "Product expected and ProductCore provided" :/

File: src/Adapter/Category/CategoryDataProvider.php [hunk 4]
```diff
@@ -69,4 +70,32 @@ public function getAllCategoriesName($root_category = null, $id_lang = false, $a
...
+        $productCategories = $product->getCategories();
+
+        $results = [];
+        foreach ($allCategories as $category) {
+            foreach($productCategories as $productCategory) {
+                if ($productCategory == $category['id_category']) {
+                    $results[] = [
+                        'id' => $category['id_category'],
+                        'name' => $category['name'],
+                        'full-path' => 'n/a'
```
Comment (mickaelandrieu) [inline src/Adapter/Category/CategoryDataProvider.php]: to be completed with data like "Home > Women > Dresses"

File: src/PrestaShopBundle/Resources/views/Admin/Categorie/tag.html.twig [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<!--<div class="pstaggerTagsWrapper " style="display: block;">
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Categorie/tag.html.twig]: remove commented code

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<h2>{{ trans("Categories", {}, 'AdminProducts') }}</h2>
+
+{{ form_errors(form.categories) }}
+{{ form_widget(form.categories) }}
+
+
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig]: remove extra lines

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+<div id="add-categories">
+
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig]: here too

File: src/PrestaShopBundle/Resources/views/Admin/Product/form.html.twig [hunk 8]
```diff
@@ -309,44 +309,10 @@
...
-                              </div>
-                              <a href="#" class="btn btn-action open">
-                                <i class="material-icons">add</i>
-                                {{ trans('Add a category', {}, 'AdminProducts') }}
-                              </a>
-                            </div>
+                            {{ include('PrestaShopBundle:Admin/Product/Include:form-categories.html.twig', { 'form': form.step1 }) }}
                           </div>
                         </div>
+
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/form.html.twig]: remove extra line

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig [hunk 9]
```diff
@@ -0,0 +1,35 @@
+<h2>{{ trans("Categories", {}, 'AdminProducts') }}</h2>
+
+{{ form_errors(form.categories) }}
+{{ form_widget(form.categories) }}
+
+<div id="add-categories">
+    <fieldset class="form-group">
+      {{ include('PrestaShopBundle:Admin:Categorie/tag.html.twig', {'categories': categories }) }}
```
Comment (vincentbz) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig]: Tu vas te faire gronder par @AlexEven. "Categorie"-->"Category", merci^^

**Comment by vincentbz** [comment]
testé et ok, merci !

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig (hunk 9 above)
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form-categories.html.twig]: Done, thanks for report @vincentbz

**Comment by AlexEven** [comment]
Hello,

A small remark regarding wording. We try to avoid having capitalized strings (like "SAVE"), but rather have it lower case as usual, but use CSS or whatever you want to have it capitalized (text-transform: capitalize;). FMostly to avoid translations not respecting the case.

Thanks!

**Comment by mickaelandrieu** [comment]
@AlexEven you're right, this is now fixed => https://github.com/PrestaShop/PrestaShop/pull/5329

Thank you for reporting this issue :)

**Pull request merged**